src/lsm/compaction.zig

```diff
@@ -315,6 +317,25 @@ pub fn CompactionType(
             assert(compaction.io_pending == 0);
             assert(!compaction.merge_done);
 
+            // Move table optimization
+            if (compaction.grid_reservation == null) {
+                const snapshot_max = snapshot_max_for_table_input(compaction.op_min);
+                const level_b = compaction.level_b;
+                const level_a = level_b - 1;
+
+                var table_a = compaction.level_a_input.?;
+                compaction.manifest.update_table(level_a, snapshot_max, &table_a);
+                assert(table_a.snapshot_max == snapshot_max);
+
+                table_a = compaction.level_a_input.?;
+                compaction.manifest.insert_table(level_b, &table_a);
+
+                compaction.merge_done = true;
+                compaction.status = .done;
+                callback(compaction);
+                return;
```

Comment:
What else can we possibly assert here? For example, compaction range or table count?

Reply:
It would also be better if we didn't depend on the optionality of `grid_reservation` to indicate `move_table`, so that we can make the intention more clear, and in the positive.

For example, perhaps the `move_table` boolean could be a function of the compaction state, instead of introducing a local constant as we do above. This would then also allow us to assert when `grid_reservation` does exist that we are not moving the table. i.e. Both positive and negative branches for both cases.